documentation/Architecture/Overview.md

```diff
@@ -2,80 +2,105 @@
 
 This document outlines the architecture of the Microsoft Performance Toolkit SDK.
 
-For more detailed information on how to create your own project using the SDK, please view [Creating your own project](../Using-the-SDK/Creating-your-project.md). 
+For more detailed information on how to create your own project using the SDK, please view [Creating an SDK Plugin C# Project](../Using-the-SDK/Creating-your-project.md). 
 
-Examples:
-- [Custom Data Source Example](../../samples/SimpleDataSource/SimpleCustomDataSource.cs).
-- [Custom Data Processor Example](../../samples/SimpleDataSource/SimpleCustomDataProcessor.cs). 
 
+## High-Level Interfaces
 
-# Microsoft Performance Toolkit SDK
+The Microsoft Performance Toolkit SDK was built to empower users to analyze arbitrary data source (such as `.etl` files, `.ctf` files, SQL server logs, etc). 
+Users can develop *SDK plugins* that contain logic for processing data sources into tables, which are sent to the *SDK driver* (such as 
```

Comment:
Wonder if we should tweak this sentence like:

... that contain logic for processing data sources into data extensions and tables

Want to call out that tables are not necessarily required.

Reply:
"Processing into data extensions" does not make sense to me. I also talk about tables being optional later. 

I will change this to say "Users can develop extensible SDK plugins" to hint that plugins can be extended. How does that sound?

Reply:
Sounds good!